src/Icicle/Avalanche/Statement/Simp/Constructor.hs

```diff
@@ -96,28 +127,51 @@ constructor a_fresh statements
    , Just (_,_,_,_,b) <- fromSum x'
    = b
 
-   | Just (PrimUnsafe (PrimUnsafeArrayIndex _), [XVar _ n, ix]) <- takePrimApps x
-   , Just x' <- get env n
-   , Just (ta, tb, a, b) <- fromZippedArray x'
+   -- update
+   | Just (PrimUpdate (PrimUpdateArrayPut _), [arr,ix,v]) <- takePrimApps x
+   , Just arrx              <- resolve env arr
+   , Just (ta, tb, i, a, b) <- fromSummedArray arrx
+   , Just vx                <- resolve env v
+   , Just (_,_,boool,l,r)   <- fromSum vx
+   , tai                    <- ArrayT BoolT
+   , taa                    <- ArrayT ta
+   , tab                    <- ArrayT tb
+   = case boool of
+      XValue _ _ (VBool False)
```

Comment:
umm. this is weird, you're doing a case on the runtime tag
maybe it just needs to be something like
```
( put i ix boool , (put a ix l , put b ix r ) )
```